_posts/2018-04-09-synching-sia.md

```diff
@@ -0,0 +1,131 @@
+
+I can sync Sia from a block height of 0 to 149000 in 73 minutes.  Here’s how I do it.
+
+## Zero subscribers
+
+As mentioned in my previous article [Benchmarking Sia’s Blockchain Sync](https://blog.spaceduck.io/sia-blockchain-sync/) Sia spends an average of 55 minutes or 19.1% of its time notifying subscribers.  I submitted the pull request [#2809](https://github.com/NebulousLabs/Sia/pull/2809) so Sia only notifies subscribers if there are any.  By running Sia with only the `consensus` and `gateway` modules loaded there will be zero subscribers.
+
+My `siad` command with minimal modules loaded.
+
+```shell
+./siad -M cg -d /root/Sia/
+```
+
+With zero subscribers Sia skips the notify subscribers functions and shaves 55 minutes off the synchronization time.  Note that with only the `consensus` and `gateway` modules loaded you cannot rent, host, or use the Sia wallet.
+
+## Reduce RPC Timeouts
+
+Sia has long timeouts when waiting for blocks to download.  I’ve submitted the pull request [#2922](https://github.com/NebulousLabs/Sia/pull/2922) to reduce some of the long timeouts.  In order to sync in 73 minutes, I dropped the timeouts in [syncronize.go](https://github.com/NebulousLabs/Sia/blob/master/modules/consensus/synchronize.go) **WAY** down.  I first tried setting `relayHeaderTimeout`, `sendBlkTimeout` and `sendBlocksTimeout` to 2 seconds, this was too low, I was timing out on all my peers and couldn’t download any blocks.  After bumping `sendBlkTimeout` and `sendBlocksTimeout` to 10 seconds I was able to stay connected to most peers and download the blocks very quickly.  Any peers that could not deliver my batch of 10 blocks in 10 seconds or less was disconnected and Sia would look for new peers.
+
+Note that 10-second timeouts are not very friendly to the Sia network and 10 seconds is not feasible for users with slower internet connections.  For example, the average Tor user or 3rd world Internet user would not be able to sync Sia with `sendBlocksTimeout` set to 10 seconds.  For this reason, the default `sendBlocksTimeout` in my pull request has been set at 180 seconds to allow for users with slower internet connections.  The Sia devs and I think this is a good balance of accommodating users with slower connections, while not wasting too much time waiting for unresponsive peers.
+
+My 73-minute sync only spent 168 seconds downloading blocks from peers.  This number is actually slightly confusing to me since if Sia downloaded 10GB of blocks it exceeded my advertised home internet speed.  A couple of things could have happened here:
+The downloaded blocks could be less than 10GB in size when sent over the Internet, I never measured how much data I actually received from my NIC.
+I could have a mistake in my timing function causing the elapsed download time to be reported wrong.
+My ISP could allow me to burst into higher speed tiers during low network congestion.
```

Comment:
Yeah, I'm skeptical of this result. That's equivalent to a download rate of 476 Mbps. Sia actually *can* go faster than most applications (including browser speed tests) because it's generating so many TCP connections to peers behind a variety of ISPs, so there are fewer links to congest (BitTorrent has the same property). But 476 just seems so suspiciously high.

Can we cut this paragraph entirely? The previous article was about breaking down the sync time into components, so it would be relevant in an article like that. But this article is just about the final 73 minute number, which we're confident in. It feels like a digression to talk about this suspicious sub-metric and speculate about it.